tests/pocket_test.cpp

```diff
@@ -0,0 +1,832 @@
+        }
+
+        THEN( "item can be successfully inserted" ) {
+            expect_can_insert( pocket_holster, glock );
+        }
+    }
+
+    GIVEN( "holster has not enough weight capacity" ) {
+        // Reduce perfect holster weight capacity by 1 gram
+        data_holster.max_contains_weight = glock.weight() - 1_gram;
+        item_pocket pocket_holster( &data_holster );
+
+        THEN( "it cannot contain the item, because it is too heavy" ) {
+            expect_cannot_contain( pocket_holster, glock, "item is too heavy",
+                                   item_pocket::contain_code::ERR_TOO_HEAVY );
+        }
+
+        THEN( "item cannot be successfully inserted" ) {
+            expect_cannot_insert( pocket_holster, glock, "item is too heavy",
+                                  item_pocket::contain_code::ERR_TOO_HEAVY );
+        }
+    }
+
+    GIVEN( "holster has not enough volume capacity" ) {
+        // Reduce perfect holster volume capacity by 1 ml
+        data_holster.volume_capacity = glock.volume() - 1_ml;
+        item_pocket pocket_holster( &data_holster );
+
+        THEN( "it cannot contain the item, because it is too big" ) {
+            expect_cannot_contain( pocket_holster, glock, "item too big",
+                                   item_pocket::contain_code::ERR_TOO_BIG );
+        }
+
+        THEN( "item cannot be successfully inserted" ) {
+            expect_cannot_insert( pocket_holster, glock, "item too big",
+                                  item_pocket::contain_code::ERR_TOO_BIG );
+        }
+    }
+
+    // TODO: Add test for item with longest_side greater than max_item_length
+
+    GIVEN( "holster already contains an item" ) {
+        // Put another item in the holster first
+        item other_glock( "test_glock" );
+        item_pocket pocket_holster( &data_holster );
+        pocket_holster.insert_item( other_glock );
+
+        THEN( "it cannot contain the item, because holster can only hold one item" ) {
+            expect_cannot_contain( pocket_holster, glock, "holster already contains an item",
+                                   item_pocket::contain_code::ERR_NO_SPACE );
+        }
+
+        THEN( "another item cannot be successfully inserted" ) {
+            expect_cannot_insert( pocket_holster, glock, "holster already contains an item",
+                                  item_pocket::contain_code::ERR_NO_SPACE );
+        }
+    }
+}
+
+// Watertight pockets
+// ------------------
+// To contain liquids, a pocket must have the "watertight" attribute set to true. After a pocket has
+// been partially filled with a liquid, adding a different liquid to it is prohibited, since liquids
+// can't mix.
+//
+// Related JSON fields:
+// "watertight"
+//
+// Functions:
+// item_pocket::watertight
+//
+TEST_CASE( "pockets containing liquids", "[pocket][watertight][liquid]" )
+{
+    // Liquids (1 charge each)
+    item ketchup( "ketchup", 0, item::default_charges_tag{} );
```

Comment:
the comment says 1 charge, but you are initializing it with default charges. which is it?

Reply:
Fixed now.